File: WebAPI/ClientApp/src/app/models/AdvertModelInfo.ts
import { AvailableTimeModel } from "./AvailableTimeModel";
import { ImageModel } from "./ImageModel";

export class AdvertModelInfo {
    public id: number;
    public name: string;
    public dateIssue: Date;
    public pts: string;
    public vin: string;
    public description: string;
    public publishDate: Date;
    public editDate: Date;
    public price: number
    public userName: string;
    public images: ImageModel[];
    public availableTimes: AvailableTimeModel[];

    public constructor(_id: number, _name: string, _dateIssue: Date, _pts: string, _vin: string, _description: string, _publishDate: Date, _editDate: Date, _price: number, _userName: string, _images: ImageModel[], _availableTimes: AvailableTimeModel[]) {
        this.id = _id;
        this.name = _name;
        this.dateIssue = _dateIssue;
        this.pts = _pts;
        this.vin = _vin;
        this.publishDate = _publishDate;
        this.editDate = _editDate;
        this.description = _description;
        this.price = _price;
        this.userName = _userName;
        this.images = _images;
        this.availableTimes = _availableTimes;
    }
}

File: WebAPI/Models/AdvertModelInfo.cs
using System;
using System.Collections.Generic;

namespace WebAPI.Models
{
    public class AdvertModelInfo
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime DateIssue { get; set; }
        public string PTS { get; set; }
        public string VIN { get; set; }
        public string Description { get; set; }
        public DateTime PublishDate { get; set; }
        public DateTime EditDate { get; set; }
        public int Price { get; set; }
        public string UserName { get; set; }
        public List<ImageModel> Images { get; set; }
        public List<AvailableTimeModel> AvailableTimes { get; set; }
    }
}
